app.js

```diff
@@ -1,18 +1,73 @@
-require('dotenv').config();
+require("dotenv").config()
 
-const express = require('express');
-const hbs = require('hbs');
+const express = require("express")
+const hbs = require("hbs")
+const SpotifyWebApi = require("spotify-web-api-node")
 
 // require spotify-web-api-node package here:
 
-const app = express();
+const app = express()
 
-app.set('view engine', 'hbs');
-app.set('views', __dirname + '/views');
-app.use(express.static(__dirname + '/public'));
+app.set("view engine", "hbs")
+app.set("views", __dirname + "/views")
+hbs.registerPartials("/views/partials")
+app.use(express.static(__dirname + "/public"))
 
 // setting the spotify-api goes here:
+const spotifyApi = new SpotifyWebApi({
+	clientId: process.env.CLIENT_ID,
+	clientSecret: process.env.CLIENT_SECRET,
+})
+
+// Retrieve an access token
+spotifyApi
+	.clientCredentialsGrant()
+	.then((data) => spotifyApi.setAccessToken(data.body["access_token"]))
+	.catch((error) => console.log("Something went wrong when retrieving an access token", error))
+
+// Business Logic
 
 // Our routes go here:
 
-app.listen(3000, () => console.log('My Spotify project running on port 3000 🎧 🥁 🎸 🔊'));
+app.get("/", (req, res) => {
+	res.render("index", { page: "Homepage" })
+})
+
+/* app.get("/artist-search", (req, res) => {
+	res.render("artist-search-results", { artists: "Kev" })
+}) */
+
+app.get("/artist-search", async (req, res) => {
+	let getArtist = await spotifyApi
+		.searchArtists(req.query.name)
+		.then((val) => {
+			return val.body.artists.items
+		})
+		.catch((err) => console.log("Something went wrong by getting the Artists: ", err))
+	console.log(getArtist)
+	res.render("artist-search-results", { artists: getArtist })
+})
+
+app.get("/albums/:id", async (req, res) => {
+	console.log()
+	await spotifyApi
+		.getArtistAlbums(req.params.id)
+		.then((val) => {
+			console.log(val.body.items)
+			res.render("albums", { albums: val.body.items })
+		})
+		.catch((err) => console.log("Something went wrong by getting the Albums: ", err))
+})
+
```

Comment:
I love that you try to implement Async/Await function.
It is an alternative to .then, and .catch, in your code you combine both.
You could have done this for example:
app.get("/artist-search", async (req, res) => {
  try {
    let allArtists = await spotifyApi.searchArtists(req.query.name);
    console.log(allArtists);
    await res.render("artist-search-results", {
      artists: allArtists.body.artists.items,
    });
  } catch (e) {
    console.error(e);
  }
});

Reply:
Ohh I thought the .then and .catch is similar to try{} catch{}